### PR TITLE
feat(function): implement pivot and unpivot

### LINKED ISSUE #10498: Feature: add pivot/unpivot function like duckdb
**Summary**
pivot move many rows value into many columns, regards values as columns names
unpivot move many columns  value into many rows , regards columns names  as values  
Description for this feature.

ref https://github.com/duckdb/duckdb/pull/6387#issuecomment-1462364955

### PR DESCRIPTION
**Pull request by SkyFan2002**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

### Syntax
The basic syntax of pivot and unpivot is currently supported:

pivot:
```sql
SELECT select_list FROM table_name PIVOT (aggregate_function(column_name) FOR column_name IN (value1, value2, ...))
[GROUP BY column_name, ...]
```
`group by` is optional, by default this is all columns not mentioned in the above clauses

unpivot:
```sql
SELECT * FROM table_name UNPIVOT(column_name FOR column_name IN (column_name..))
```

### Implement
Like duckdb, pivoting and unpivoting are implemented entirely as rewrites into SQL queries. Specifically, it is implemented by rewriting `SelectStmt` in the bind stage.

Now unpivot works fine, pivot results are the same as equivalent SQL statements, but due to agg_if implementation seems to have bugs, pivot results are not correct.

For example,the following two statements are equivalent:
```sql
query IIIII
SELECT *
  FROM monthly_sales
    PIVOT(SUM(amount) FOR MONTH IN ('JAN', 'FEB', 'MAR', 'APR'))
  ORDER BY EMPID;
----
1	10400	8000	11000	18000
2	39500	90700	12000	5300

query IIIII
SELECT EMPID,
    SUM_IF(AMOUNT,MONTH = 'JAN') AS JAN,
    SUM_IF(AMOUNT,MONTH = 'FEB') AS FEB,
    SUM_IF(AMOUNT,MONTH = 'MAR') AS MAR,
    SUM_IF(AMOUNT,MONTH = 'APR') AS APR
    FROM monthly_sales
    GROUP BY EMPID;
----
1	10400	8000	11000	18000
2	39500	90700	12000	5300
```

Their results of them are indeed the same, but they are not correct：
```
1 10400 NULL NULL NULL                                                                                              +
2 39500 NULL NULL NULL
```
Closes #10498

### CODE REVIEW

File: scripts/ci/ci-run-sqllogic-tests.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ echo "Run suites using argument: $RUN_DIR"
 
 for i in $(seq 1 3); do
 	echo "Starting databend-sqllogic tests $i"
-	target/${BUILD_PROFILE}/databend-sqllogictests --handlers ${TEST_HANDLERS} ${RUN_DIR} --skip_dir management --enable_sandbox --parallel 8 --debug
+	target/${BUILD_PROFILE}/databend-sqllogictests --handlers ${TEST_HANDLERS} ${RUN_DIR} --skip_dir management,base,crdb,debug,mode,query,tpch,ydb,duckdb --enable_sandbox --parallel 4 --debug
```
[the remark this answers is not in the document]
Comment (sundy-li) [reply scripts/ci/ci-run-sqllogic-tests.sh]: BTW, logictests can run by file.  `--run_file`

File: src/query/ast/src/ast/query.rs [hunk 2]
```diff
@@ -164,6 +202,20 @@ pub enum TimeTravelPoint {
     Timestamp(Box<Expr>),
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct PivotMeta {
```
Comment (b41sh) [inline src/query/ast/src/ast/query.rs:206]: How about using `Povit` for the name?

File: src/query/ast/src/ast/query.rs [hunk 3]
```diff
@@ -164,6 +202,20 @@ pub enum TimeTravelPoint {
...
+    pub aggregate: Expr,
+    pub pivot_column: Identifier,
+    pub pivot_values: Vec<Expr>,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct UnpivotMeta {
```
Comment (b41sh) [inline src/query/ast/src/ast/query.rs:213]: ditto

File: src/query/ast/src/ast/format/ast_format.rs [hunk 4]
```diff
@@ -2193,6 +2193,8 @@ impl<'ast> Visitor<'ast> for AstFormatVisitor {
                 table,
                 alias,
                 travel_point,
+                pivot: _,
+                unpivot: _,
```
Comment (b41sh) [inline src/query/ast/src/ast/format/ast_format.rs:2197]: `pivot` and `unpivot` need to be processed for the display of `explain ast`

File: src/query/ast/src/ast/format/syntax/query.rs [hunk 5]
```diff
@@ -275,6 +275,8 @@ pub(crate) fn pretty_table(table: TableReference) -> RcDoc<'static> {
             table,
             alias,
             travel_point,
+            pivot: _,
+            unpivot: _,
```
Comment (b41sh) [inline src/query/ast/src/ast/format/syntax/query.rs:279]: ditto

**Review by sundy-li — APPROVED** [review]

**Pull request merged**